Content.Shared/Magic/SharedMagicSystem.cs

```diff
@@ -542,6 +543,34 @@ private void OnRandomGlobalSpawnSpell(RandomGlobalSpawnSpellEvent ev)
         _audio.PlayGlobal(ev.Sound, ev.Performer);
     }
 
+    #endregion
+    #region Mindswap Spells
+
+    private void OnMindSwapSpell(MindSwapSpellEvent ev)
+    {
+        if (ev.Handled || !PassesSpellPrerequisites(ev.Action, ev.Performer))
+            return;
+
+        ev.Handled = true;
+        Speak(ev);
+
+        // Need performer mind, but target mind is unnecessary, such as taking over a NPC
+        // Need to get target mind before putting performer mind into their body if they have one
+        // Thus, assign bool before first transfer, then check afterwards
+
+        if (!_mind.TryGetMind(ev.Performer, out var perMind, out var perMindComp))
+            return;
+
+        var tarHasMind = _mind.TryGetMind(ev.Target, out var tarMind, out var tarMindComp);
+
+        _mind.TransferTo(perMind, ev.Target);
+
+        if (tarHasMind)
+        {
+            _mind.TransferTo(tarMind, ev.Performer);
+        }
+    }
+
```

Comment:
We should add some sort of knockout time here.
I feel like 30 seconds is a bit too long. It does force a wizard to swap with someone alone and not in a team fight, since it'll be a dead giveaway. And the idea of a mouse wizard is also hilarious. So let's try a 10 second stun first to see how it goes.